packages/worker/src/worker.ts

```diff
@@ -821,21 +918,94 @@ export class Worker {
    * Errors from core responses are translated to cancellation requests and fed back via the activityFeedbackSubject.
    */
   protected activityHeartbeat$(): Observable<void> {
+    function process(state: HeartbeatState, heartbeat: Heartbeat): HeartbeatStateAndOutput {
+      return { state: { ...state, processing: true, pending: undefined }, output: { type: 'send', heartbeat } };
+    }
+
+    function storePending(state: HeartbeatState, heartbeat: Heartbeat): HeartbeatStateAndOutput {
+      return { state: { ...state, pending: heartbeat }, output: null };
```

Comment:
Is it possible we'd ever need to store multiple pending heartbeats?

Reply:
No, the whole point is that we only flush the last one because the others become obsolete